build_compilers/lib/src/modules.dart

```diff
@@ -5,26 +5,55 @@
 import 'package:analyzer/dart/element/element.dart';
 import 'package:build/build.dart';
 
+/// A collection of Dart libraries in a strongly connected component and the
+/// modules they depend on.
+///
+/// Modules can span pub package boundaries when there are import cycles across
+/// packages.
 class Module {
-  /// The libraries in the strongly connected cycle.
+  /// The library which will be used to reference any library in [sources].
   ///
-  /// In most cases without cyclic imports this will contain a single asset.
-  final Set<AssetId> cycle;
+  /// The assets which are built once per module, such as DDC compiled output or
+  /// Analyzer summaries, will be build for the primary source and not for any
```

Comment:
This makes it sound as if the other sources aren't included in the summary. Maybe instead say that the summaries/ddc output will be named after the primary source?

Reply:
Rephrased.